### DISCUSSION TITLE
Why doesn't `packages/ui` have a `.eslintrc.js` file whereas `apps/docs` and `apps/web` do? And why do they need one?

### DISCUSSION THREAD

**Discussion by albertothedev** [vercel/turborepo#3381] [Help] [answered]
In the default Turborepo configuration, three directories are set up to be linted: `apps/docs`, `apps/web`, and `packages/ui`. However, `packages/ui` is the only one without a `.eslintrc.js` file. This means there are two separate `.eslintrc.js` files in three different places:

`.eslintrc.js` (root):
```javascript
module.exports = {
  root: true,
  // This tells ESLint to load the config from the package `eslint-config-custom`
  extends: ["custom"],
  settings: {
    next: {
      rootDir: ["apps/*/"],
    },
  },
};
```

`apps/docs` and `apps/web`:
```javascript
module.exports = {
  root: true,
  extends: ["custom"],
};
```
[The documentation reads](https://turbo.build/repo/docs/getting-started/create-new#understanding-eslint-config-custom):

```
The way ESLint finds its configuration file is by looking at the closest .eslintrc.js. If it can't find one in the current directory, it'll look in the directory above until it finds one.

So that means that if we're working on code inside packages/ui (which doesn't have a .eslintrc.js) it'll refer to the root instead.

Apps that do have an .eslintrc.js can refer to custom in the same way. 
```

My question is: 

If the `.eslintrc.js` file on the root is extending the custom configuration and [specifying that the `eslint-plugin-next` package can be found at the root of the `apps` directory](https://nextjs.org/docs/basic-features/eslint#rootdir), doesn't that mean that the `.eslintrc.js` files in `apps/docs` and apps/web, which also extend the custom configuration, are redundant? If I have understood the documentation right, it seems the linting would work the same without them.

**Comment by mehulkar** [comment]
It's possible this is an oversight?  cc @tknickman 
https://github.com/vercel/turbo/tree/main/packages/create-turbo/templates/_shared_ts

**Reply by albertothedev** [reply]
Are you suggesting that the `.eslintrc.js` files in `apps/docs` and `apps/web` might indeed be redundant?

**Reply by mehulkar** [reply]
Yes. Although they also set `root: true`, so it's possible they are there to make the workspaces isolated (and _not_ inherit from the root eslint config). I don't think it's wrong that they're there, but generally speaking eslint is also wanting to move towards a flat file config, so I would think that's what we want to do as well. 

Can you make a PR and we can discuss more concretely?

**Reply by albertothedev** [reply]
That's what I don't get. What's the point of making the workspace independant like that when it's just extending the custom configuration? At least in the starter template's use case.

Would you like me to create a PR removing both `.eslintrc.js` files in `apps/docs` and `apps/web`?

**Reply by tknickman** [reply]
This was implemented to demonstrate different methods of eslint config inheritance - we left both in place with the docs explanation to make both options available depending on your use case.

**Reply by albertothedev** [reply]
Where's the part of the docs that explains that they are two ways of doing the same thing and that it's made that way to illustrate them? And I still don't understand what the difference is between them, which is something that the docs should explain too and that I don't think they do. What use cases would benefit from either? The explanation should be in the "Getting started" section, as that is the section that most concerns the setup of the starter template.

**Reply by mehulkar** [reply]
@albertothedev what you're saying makes sense, thanks for the attention to detail. 
We want the examples to make sense and also work out of the box. Happy to take that PR!

**Reply by thedaviddias** [reply]
I have been painfully trying to understand the different ways to configure eslint in a monorepo. I follow @albertothedev comment, where it would be really helpful if those questions could be answered somehow.

**Reply by albertothedev** [reply]
@thedaviddias The comments on this PR I opened: https://github.com/vercel/turbo/pull/4172 give more information on the topic. You might find them helpful.

**Reply by thedaviddias** [reply]
Awesome @albertothedev, I'm gonna take a look! Appreciate the answer.

**Comment by tknickman** [comment, marked as answer]
Apologies for the confusion here @albertothedev. The way this is setup in the template is a little bit confusing and unnecessary. This is meant to demonstrate a situation when, to take one example, you have a monorepo comprised of all react apps / packages, and have added a single _different_ application (Next.js, Vue etc). You may want that single eslint config at root that applies to everything, but then override an individual app or package with its own config that has more specific plugins etc. 

But in its current form, this _is_ a bit confusing - PR's welcome!

**Comment by albertothedev** [comment]
I have opened the RFC #3621 to discuss solutions to this issue.
